### PR TITLE
UXG-1298: Adding common actions and mutations with progress stepper

### PR DESCRIPTION
**Pull request by sgreenl5**
UXG-1298: Added in the progress stepper state and updated the current views to use it. Also I added in common action/mutation so that there is a single source of truth for those instead of repeated code. Updated the forms to match.

### CODE REVIEW

File: src/vue/backend/src/store/modules/progressStepper/actions.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import * as types from '../../common/mutationTypes'
+
+const progressStepperUpdate = ({ commit, state }, payload) => {
+    // Make a deep copy of the state to maintain name, total, and icon
+    // Replace the active step with the new active step
+    const updatedSteps = state.guidedSetupSteps.map((step, key) => {
+        const newStep = {
```
Comment (ralls) [inline src/vue/backend/src/store/modules/progressStepper/actions.js:7]: Not necessary to declare a const - just return the object

File: src/vue/backend/src/store/index.js [hunk 2]
```diff
@@ -15,12 +21,24 @@ const modules = {
...
+const bulkApplyCommonPieces = (storeModules) => {
+    return Object.keys(storeModules).reduce((acc, cur) => {
+        acc[cur] = applyCommonPieces(storeModules[cur])
+        return acc
+    }, {})
+}
 const store = new Vuex.Store({
     strict: process.env.NODE_ENV !== 'production',
-    modules
+    modules: bulkApplyCommonPieces(modules)
```
Comment (ralls) [inline src/vue/backend/src/store/index.js:41]: This is great! 👍

**Review by ralls — DISMISSED** [review]
Looks good to me 👍

**Review by nicmarson — APPROVED** [review]
Love seeing code getting more efficient, looks great!

**Review by ralls — APPROVED** [review]
👍

**Pull request merged**